k2/python/tests/ragged_tensor_test.py

```diff
@@ -227,6 +228,55 @@ def test_sum_no_grad(self):
 
                 assert torch.all(torch.eq(b, expected_sum))
 
+    def test_logsumexp_with_grad(self):
+        for device in self.devices:
+            for dtype in self.real_dtypes:
+                a = k2r.RaggedTensor([[-1.1, 2.2, -3.3, 4.4], [], [5.5, -6.6]],
+                                     dtype=dtype, device=device)
+                a_0 = torch.tensor([-1.1, 2.2, -3.3, 4.4],
+                                   dtype=dtype, device=device)
+                a_1 = torch.tensor([], dtype=dtype, device=device)
+                a_2 = torch.tensor([5.5, -6.6], dtype=dtype, device=device)
+
+                a.requires_grad_(True)
+                a_0.requires_grad_(True)
+                a_1.requires_grad_(True)
+                a_2.requires_grad_(True)
+
+                b = a.logsumexp()
+                b_0 = a_0.logsumexp(dim=0)
+                b_1 = a_1.logsumexp(dim=0)
+                b_2 = a_2.logsumexp(dim=0)
+
+                expected_b = torch.stack([b_0, b_1, b_2])
+                assert torch.allclose(b, expected_b)
+
+                c = b.sum()
+                expected_c = expected_b.sum()
+                c.backward()
+                expected_c.backward()
+                assert torch.allclose(a.grad,
+                                      torch.cat([a_0.grad, a_1.grad, a_2.grad]))
+
+    def test_logsumexp_no_grad(self):
```

Comment:
This function can be merged with the above function to reduce the amount of test code.